Source/WebKit/Scripts/webkit/tests/TestWithEnabledByMessageReceiver.cpp

```diff
@@ -42,22 +42,35 @@ void TestWithEnabledBy::didReceiveMessage(IPC::Connection& connection, IPC::Deco
 {
     auto& sharedPreferences = sharedPreferencesForWebProcess(connection);
     UNUSED_VARIABLE(sharedPreferences);
+    bool runtimeEnablementCheckFailed = false;
+    UNUSED_VARIABLE(runtimeEnablementCheckFailed);
     Ref protectedThis { *this };
     if (decoder.messageName() == Messages::TestWithEnabledBy::AlwaysEnabled::name())
         return IPC::handleMessage<Messages::TestWithEnabledBy::AlwaysEnabled>(connection, decoder, this, &TestWithEnabledBy::alwaysEnabled);
-    if (decoder.messageName() == Messages::TestWithEnabledBy::ConditionallyEnabled::name() && sharedPreferences.someFeature)
-        return IPC::handleMessage<Messages::TestWithEnabledBy::ConditionallyEnabled>(connection, decoder, this, &TestWithEnabledBy::conditionallyEnabled);
-    if (decoder.messageName() == Messages::TestWithEnabledBy::ConditionallyEnabledAnd::name() && (sharedPreferences.someFeature && sharedPreferences.otherFeature))
-        return IPC::handleMessage<Messages::TestWithEnabledBy::ConditionallyEnabledAnd>(connection, decoder, this, &TestWithEnabledBy::conditionallyEnabledAnd);
-    if (decoder.messageName() == Messages::TestWithEnabledBy::ConditionallyEnabledOr::name() && (sharedPreferences.someFeature || sharedPreferences.otherFeature))
-        return IPC::handleMessage<Messages::TestWithEnabledBy::ConditionallyEnabledOr>(connection, decoder, this, &TestWithEnabledBy::conditionallyEnabledOr);
+    if (decoder.messageName() == Messages::TestWithEnabledBy::ConditionallyEnabled::name()) {
+        if (sharedPreferences.someFeature)
+            return IPC::handleMessage<Messages::TestWithEnabledBy::ConditionallyEnabled>(connection, decoder, this, &TestWithEnabledBy::conditionallyEnabled);
+        runtimeEnablementCheckFailed = true;
```

Comment:
It seems more straight forward to just directly call `connection.markCurrentlyDispatchedMessageAsInvalid()` here directly instead of setting a boolean then calling `markCurrentlyDispatchedMessageAsInvalid()` conditionally.

Reply:
The marking needs to go after `ignoreInvalidMessageForTesting` check, since we don't want to terminate process in testing.

Reply:
oh I see. Hm... maybe it's better to check `connection.ignoreInvalidMessageForTesting` and call `connection.markCurrentlyDispatchedMessageAsInvalid` conditionally? ignoreInvalidMessageForTesting() is a single boolean check so it shouldn't bloat the code size too much.

Reply:
Thinking about this, actually this would add multiple lines for each annotated message because `connection.ignoreInvalidMessageForTesting()` is only available at `ENABLE(IPC_TESTING_API)`, so it will be like:
```
#if ENABLE(IPC_TESTING_API)
    if (connection.ignoreInvalidMessageForTesting())
        return;
#endif // ENABLE(IPC_TESTING_API)
    return connection.markCurrentlyDispatchedMessageAsInvalid();
```

So I am going to upload the a new patch with current implementation; let me know if you still prefer adding the lines to each message.

Reply:
Hm... I wonder if we should make `markCurrentlyDispatchedMessageAsInvalid()` automatically respect `ignoreInvalidMessageForTesting()`? That'll simply this and a bunch of other code. It's fine to land your version as is but it would be nice if we didn't have to introduce a boolean state variable.